lib/shopify_api/resources/base.rb

```diff
@@ -57,21 +71,31 @@ def activate_session(session)
         raise InvalidSessionError.new("Session cannot be nil") if session.nil?
         self.site = session.site
         self.headers.merge!('X-Shopify-Access-Token' => session.token)
+        self.api_version = session.api_version
       end
 
       def clear_session
         self.site = nil
         self.password = nil
         self.user = nil
+        self.api_version = nil
         self.headers.delete('X-Shopify-Access-Token')
       end
 
-      def api_prefix
-        API_PREFIX
+      def api_version
+        if _api_version_defined?
+          _api_version
+        elsif superclass != Object && superclass.site
```

Comment:
What's going on here?

Reply:
pulled from older versions of active resource and how the to correctly pull the value from the parent class and not have the value changed.
https://github.com/rails/activeresource/blob/v4.0.0/lib/active_resource/base.rb#L418-L439